be/test/storage/lake/schema_change_test.cpp

```diff
@@ -877,4 +877,242 @@ INSTANTIATE_TEST_SUITE_P(SchemaChangeModifyColumnOrderTest, SchemaChangeModifyCo
+        ASSERT_OK(handler.process_alter_tablet(request));
+    }
+
+    for (int i = 0; i < GetParam().writes_after; i++) {
+        VChunk chunk1({ck1, ck0, cv0}, _new_schema);
+
+        ASSIGN_OR_ABORT(auto delta_writer, DeltaWriterBuilder()
+                                                   .set_tablet_manager(_tablet_manager.get())
+                                                   .set_tablet_id(new_tablet_id)
+                                                   .set_txn_id(txn_id)
+                                                   .set_partition_id(_partition_id)
+                                                   .set_mem_tracker(_mem_tracker.get())
+                                                   .build());
+        ASSERT_OK(delta_writer->open());
+        ASSERT_OK(delta_writer->write(chunk1, indexes.data(), indexes.size()));
+        ASSERT_OK(delta_writer->finish());
+        delta_writer->close();
+        ASSERT_OK(_tablet_manager->publish_log_version(new_tablet_id, txn_id, version + 1));
+        version++;
+        txn_id++;
+    }
+
+    ASSERT_OK(_tablet_manager->publish_version(new_tablet_id, 1, version + 1, &alter_txn_id, 1).status());
+    version++;
+    txn_id++;
+
+    // check new tablet data
+    std::vector<int> rc1_0{1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3};
+    std::vector<int> rc0_0{1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4};
+    std::vector<int> rc2_0{2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24};
+
+    ASSIGN_OR_ABORT(auto new_tablet, _tablet_manager->get_tablet(new_tablet_id));
+    ASSIGN_OR_ABORT(auto reader, new_tablet.new_reader(version, *_new_schema));
+    CHECK_OK(reader->prepare());
+    CHECK_OK(reader->open(TabletReaderParams()));
+
+    auto chunk = ChunkHelper::new_chunk(*_new_schema, 1024);
+
+    if (GetParam().writes_before + GetParam().writes_after > 0) {
+        CHECK_OK(reader->get_next(chunk.get()));
+        for (int i = 0, sz = k0.size(); i < sz; i++) {
+            EXPECT_EQ(rc1_0[i], chunk->get(i)[0].get_int32());
+            EXPECT_EQ(rc0_0[i], chunk->get(i)[1].get_int32());
+            EXPECT_EQ(rc2_0[i], chunk->get(i)[2].get_int32());
+        }
+        chunk->reset();
+    }
+
+    auto st = reader->get_next(chunk.get());
+    ASSERT_TRUE(st.is_end_of_file());
+}
+
+// clang-format off
+INSTANTIATE_TEST_SUITE_P(SchemaChangeSortKeyReorderTest, SchemaChangeSortKeyReorderTest,
+                         ::testing::Values(SchemaChangeParam{PRIMARY_KEYS, 0, 0},
+                                           SchemaChangeParam{PRIMARY_KEYS, 0, 1},
+                                           SchemaChangeParam{PRIMARY_KEYS, 0, 2},
+                                           SchemaChangeParam{PRIMARY_KEYS, 2, 0},
+                                           SchemaChangeParam{PRIMARY_KEYS, 2, 1},
+                                           SchemaChangeParam{PRIMARY_KEYS, 2, 2},
+                                           SchemaChangeParam{PRIMARY_KEYS, 2, 4}),
```

Comment:
Maybe it's not necessary to test so many different data sizes, they don't make much sense for schema change.